feat: :sparkles: add template README in `docs/` (#136)
seedcase-project/template-data-package: template/docs/README.md

Author: I was aiming for a medium level of detail -- let me know if it should be more or less!

    checker configuration file.
- `CITATION.cff`: Structured citation metadata for your project.
- `justfile`: [`just`](https://just.systems/man/en/) configuration
    file for scripting project tasks.
- `main.py`: Central script file for the Data Package. This is where
    helper scripts are invoked and work together to create and manage
    the Data Package.
- `pyproject.toml`: Main Python project configuration file defining
    metadata and dependencies.
- `README.md`: Autogenerated description of the Data Package. Not a
    development guide. Information on using and developing the project
    should be included in `docs/README.md`.
- `ruff.toml`: [Ruff](https://docs.astral.sh/ruff/) configuration file
    for linting and formatting Python code.
- `uv.lock`: Lockfile used by [`uv`](https://docs.astral.sh/uv/) to
    record exact versions of installed dependencies.

## How to develop your Data Package

In your new project generated from the `template-data-package`, the
first steps for creating and developing your Data Package are already
set up in `main.py`. For more detailed instructions on using Seedcase
Sprout to organise your Data Package, see the

> [guides](https://sprout.seedcase-project.org/docs/guide/) on Sprout's

Reviewer: ```suggestion
[guide](https://sprout.seedcase-project.org/docs/guide/) on Sprout's
```

website. You can read more about the files and folders created by
`main.py` on the
[Outputs](https://sprout.seedcase-project.org/docs/design/interface/outputs)
page of the design documentation.

### Creating package properties

1. Run `main.py` to create the `scripts/package_properties.py` file for
    the properties of your Data Package.

>     ``` {.bash filename="Terminal"}

Reviewer: ```suggestion
    ``` bash
```

Since we don't yet have this template use Quarto, we can't use these `filename` add ons.

    just build
    ```

    You can also run `main.py` by clicking the "Run" button in your IDE.

2. Open `scripts/package_properties.py` and fill in all required
    fields. Also fill in any optional fields you find useful. You can
    always update these later. Make sure to save the file.

3. In `main.py`, uncomment the lines referencing the

>     `package_properties` and `package_path` variables: lines 4, 17, 29,
>     34, and 36.

Reviewer: ```suggestion
    `package_properties` and `package_path` variables.
```

This level of detail isn't necessary and also very fragile to change.

4. Rerun `main.py` to create the `datapackage.json` and `README.md`
    files for your Data Package.

### Creating a new resource

#### With data to add to the resource

> 

Reviewer: ```suggestion

While you can create resource properties without data, it is
a lot more challenging. If at all possible, only create a 
resource properties object when you have data to use to
at least pre-fill in some of the important fields.
```

> If you already have some data in a tidy format, load this as a Polars

Reviewer: ```suggestion
In order to use Sprout, the data needs to already be in a tidy format. 
When it is, load the data as a Polars
```

data frame into the `raw_data` variable in `main.py`.

> 1. Uncomment lines up to and including the creation of resource
>     properties: lines 1, 18, 20, and 23-26.

Reviewer: ```suggestion
1. Uncomment lines up to and including the creation of resource
    properties.
```

> 2. Fill in the `resource_name` argument in line 24.

Reviewer: ```suggestion
2. Fill in the `resource_name` argument.
```

3. Rerun `main.py` to create the
    `scripts/resource_properties_<name>.py` file for the properties of
    the new resource.

4. Open `scripts/resource_properties_<name>.py` and fill in all
    required fields. Also fill in any optional fields you find useful.
    You can always update these later. Make sure to save the file.

5. In `package_properties.py`, import your new resource properties by

>     uncommenting line 3 and updating it with the name of your resource.

Reviewer: ```suggestion
    uncommenting and updating it with the name of your resource.
```

>     Also uncomment the `resources` field in lines 46-48 and update the

Reviewer: ```suggestion
    Also uncomment the `resources` field and update the
```

    name of the resource properties in the array to match the name of
    your new resource.

6. In `main.py`, import your new resource properties by uncommenting

>     line 6 and updating it with the name of your resource.

Reviewer: ```suggestion
    it and updating it with the name of your resource.
```

> 7. Uncomment all remaining lines in the file and rename the

Reviewer: ```suggestion
7. Uncomment everything else in the `main.py` file and rename the
```

    `resource_properties` variable to the name of the new resource
    properties you just imported.

8. Rerun `main.py`. This will:

    - Update `datapackage.json` and `README.md`.
    - Create a `resources/` folder containing a folder for your new
        resource. In here, you will find a `batch/` folder with the
        individual data batches you've uploaded for this resource and a
        `data.parquet` file containing all resource data.

> #### Without data to add to the resource
> 
> You can create a new resource without adding data to it using a shorter
> version of the steps above.
> 
> - Step 1: uncomment only lines 23, 24, and 26.
> - Step 4: if you cannot describe columns in your data at this stage,
>     comment out the `schema` field.
> - Steps 6-7: as you have no data to add, you should skip these steps.
> 

Reviewer: ```suggestion
```

Let's not get into describing situations where they don't have data, it gets complicated and we don't have a good flow there yet.

## How to use the `justfile`

The `justfile` contains scripts or "recipes" that are shorthands for
performing common project tasks. You can get an overview of available
recipes by running

> ``` {.bash filename="Terminal"}

Reviewer: ````suggestion
``` bash
````

just
```

in the project root.

You can run a recipe by typing

> ``` {.bash filename="Terminal"}

Reviewer: ````suggestion
``` bash
````
